packages/react-server/src/ReactFlightServer.js

```diff
@@ -2202,6 +2232,9 @@ function renderModel(
 
     if (request.status === ABORTING) {
       task.status = ABORTED;
+      if (enableHalt && request.fatalError === haltSymbol) {
+        return serializeInfinitePromise();
```

Comment:
So I don't think this is actually right. Because this is saying that this value will be a Promise but if this wasn't already a Promise that's incorrect. It's not the same as postpone which gets serialized as an error. What you want here is more like `serializeByValueID` to an ID that will never resolve.